app/src/main/java/com/epfl/drawyourpath/userProfile/cache/UserModelCached.kt

```diff
@@ -66,6 +66,11 @@ class UserModelCached(application: Application) : AndroidViewModel(application)
         .map { entity -> entity?.let { UserProfile(it) } ?: UserProfile(UserEntity("empty user")) }
 
     // dailyGoal
+    private val dailyGoals: LiveData<List<DailyGoal>> = user.switchMap { user ->
+        dailyGoalCache.getDailyGoalById(user.userId).map { dailyGoalEntities ->
+            dailyGoalEntities.map { DailyGoal(it) }
+        }
+    }

```

Comment:
this could be refactored as 
```kotlin
private val dailyGoals: LiveData<List<DailyGoal>> = _currentUserID.switchMap {
        dailyGoalCache.getDailyGoalById(it)
    }.map { entities -> entities.map { DailyGoal(it) } }
```
this only needs the userId so no need for the user